File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        lookup = {}
        for i, num in enumerate(nums):
            comp = target - num
            if comp in lookup:
                return [lookup[comp], i]
            lookup[num] = i
        
        

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> lookup = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int comp = target - nums[i];
            if (lookup.containsKey(comp)) {
                int[] result = new int[]{lookup.get(comp), i};
                return result;
            }
            lookup.put(nums[i], i);
        } 
        return null;
    }
}